cattle/plugins/docker/compute.py

```diff
@@ -306,19 +340,24 @@ def to_upper_case(key):
                     if e.message.response.status_code == 404:
                         # Ensure image is pulled, somebody could have deleted
                         # it behind the scenes
-                        pull_image(instance.image, progress)
+                        pull_image(image_tag, None, auth_config=auth_config)
                         cc = create_config
-                        container = c.create_container(image_tag, **cc)
+                        try:
```

Comment:
I think this will work

```
            try:
                container = c.create_container(image_tag, **create_config)
            except APIError as e:
                if e.message.response.status_code == 404:
                    # Ensure image is pulled, somebody could have deleted
                    # it behind the scenes
                    pull_image(instance.image, progress)
                    cc = create_config
                    container = c.create_container(image_tag, **cc)
                else:
                    raise(e)
```